# Peer-Graded Assignment: Segmenting and Clustering

## Loading some data from the previous exercise

### No explanation is provided. Please refer to previous exercise

In [1]:
import pandas as pd
import numpy as np
import random
import requests

wikipedia = "https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M"
wiki_get = requests.get(wikipedia)

wiki_raw = pd.read_html(wiki_get.content,header=0)[0]
df = wiki_raw[wiki_raw.Borough != 'Not assigned']
df.reset_index(drop=True, inplace=True)
df.shape

(103, 3)

In [2]:
df.head()

Postal Code           Borough                                 Neighborhood
0         M3A        North York                                    Parkwoods
1         M4A        North York                             Victoria Village
2         M5A  Downtown Toronto                    Regent Park, Harbourfront
3         M6A        North York             Lawrence Manor, Lawrence Heights
4         M7A  Downtown Toronto  Queen's Park, Ontario Provincial Government

## The second part continues

Now that we have a database we can load the location data for that place. But before that we need to install the required packages and import them

In [3]:
!pip install geocoder

The geocoder package is inatalled and it can be loaded and the location of geodata can be specified

In [4]:
import geocoder

location = 'http://cocl.us/Geospatial_data'

Now import the data from the geo website based on postal codes

In [5]:
df_geoloc = pd.read_csv(location)
df_geoloc.head()

Postal Code   Latitude  Longitude
0         M1B  43.806686 -79.194353
1         M1C  43.784535 -79.160497
2         M1E  43.763573 -79.188711
3         M1G  43.770992 -79.216917
4         M1H  43.773136 -79.239476

In [6]:
df_geoloc.shape

(103, 3)

Here both the dataframe has the same amount of entries i.e. 103. so the frame should be stable as per my view. Now join both the data frame with join function

In [7]:
df = df.join(df_geoloc.set_index('Postal Code'), on='Postal Code')
df.head()

Postal Code           Borough                                 Neighborhood  \
0         M3A        North York                                    Parkwoods   
1         M4A        North York                             Victoria Village   
2         M5A  Downtown Toronto                    Regent Park, Harbourfront   
3         M6A        North York             Lawrence Manor, Lawrence Heights   
4         M7A  Downtown Toronto  Queen's Park, Ontario Provincial Government   

    Latitude  Longitude  
0  43.753259 -79.329656  
1  43.725882 -79.315572  
2  43.654260 -79.360636  
3  43.718518 -79.464763  
4  43.662301 -79.389494

In [8]:
df.shape

(103, 5)